daemons/fenced/fenced_commands.c

```diff
@@ -2731,12 +2755,16 @@ fence_locally(xmlNode *msg, pcmk__action_result_t *result)
 {
     const char *device_id = NULL;
     stonith_device_t *device = NULL;
-    async_command_t *cmd = create_async_command(msg);
-    xmlNode *dev = get_xpath_object("//@" F_STONITH_TARGET, msg, LOG_ERR);
+    async_command_t *cmd = NULL;
+    xmlNode *dev = NULL;
 
-    CRM_CHECK(result != NULL, return);
+    CRM_CHECK((msg != NULL) && (result != NULL), return);
 
+    dev = get_xpath_object("//@" F_STONITH_TARGET, msg, LOG_ERR);
+
+    cmd = create_async_command(msg);
     if (cmd == NULL) {
+        crm_log_xml_warn(msg, "invalid");
```

Comment:
missing a bit of the why here but if it is worth passing result into create_async_command and all that ...

Reply:
If designing from scratch it might make sense, but since we'd just use fenced_set_protocol_error() for all failure cases, probably not worth the effort now